fc/mef.py

```diff
@@ -539,54 +261,67 @@ def sc_fun(p,x):
         return np.exp(p[0] * x + p[1])
     
     # Fit parameters
-    err_par = lambda p: err_fun(p, peaks_ch, peaks_mef)
+    err_par = lambda p: err_fun(p, fl_channel, fl_mef)
     res = minimize(err_par, params, options = {'gtol': 1e-6})
 
     # Separate parameters
-    sc_params = res.x
+    beads_params = res.x
 
     # Beads model function
-    sc_beads = lambda x: fit_fun(sc_params, x)
+    beads_model = lambda x: fit_fun(beads_params, x)
 
     # Standard curve function
-    sc = lambda x: sc_fun(sc_params, x)
+    std_crv = lambda x: sc_fun(beads_params, x)
     
-    return (sc, sc_beads, sc_params)
-
-def plot_standard_curve(peaks_ch, 
-                        peaks_mef,
-                        sc_beads,
-                        sc_abs,
-                        xlim = (0., 1023.),
-                        ylim = (1, 1e8)):
+    return (std_crv, beads_model, beads_params)
```

Comment:
I don't think any of the three first points would make any difference. This is a relatively easy and smooth optimization problem. Also, the first peak seen by this function is not autofluorescence if the user decided to discard the first peak.

I don't completely understand the last point. Was this a problem that you actually saw? I don't think this is being a problem currently, otherwise running the example excel file would give the same standard curve three times.